multiqc/modules/bcl2fastq/bcl2fastq.py

```diff
@@ -0,0 +1,147 @@
+        run_data = self.bcl2fastq_data[runId]
+        for conversionResult in content["ConversionResults"]:
+            lane = 'L{}'.format(conversionResult["LaneNumber"])
+            if lane in run_data:
+                log.debug("Duplicate runId/lane combination found! Overwriting: {}".format(self.prepend_runid(runId, lane)))
+            run_data[lane] = {"total": 0, "perfectIndex": 0, "samples": dict()}
+            for demuxResult in conversionResult["DemuxResults"]:
+                sample = demuxResult["SampleName"]
+                run_data[lane]["samples"][sample] = {"total": 0, "perfectIndex": 0, "filename": os.path.join(myfile['root'],myfile["fn"])}
+                run_data[lane]["total"] += demuxResult["NumberReads"]
+                run_data[lane]["samples"][sample]["total"] += demuxResult["NumberReads"]
+                for indexMetric in demuxResult["IndexMetrics"]:
+                    run_data[lane]["perfectIndex"] += indexMetric["MismatchCounts"]["0"]
+                    run_data[lane]["samples"][sample]["perfectIndex"] += indexMetric["MismatchCounts"]["0"]
+            run_data[lane]["samples"]["undetermined"] = {"total": conversionResult["Undetermined"]["NumberReads"], "perfectIndex": 0}
+
+    def split_data_by_lane_and_sample(self):
+        for runId in self.bcl2fastq_data.keys():
+            for lane in self.bcl2fastq_data[runId].keys():
+                uniqLaneName = self.prepend_runid(runId, lane)
+                self.bcl2fastq_bylane[uniqLaneName] = {
+                "total": self.bcl2fastq_data[runId][lane]["total"],
+                "perfectIndex": self.bcl2fastq_data[runId][lane]["perfectIndex"],
+                "undetermined": self.bcl2fastq_data[runId][lane]["samples"]["undetermined"]["total"]
+                }
+                for sample in self.bcl2fastq_data[runId][lane]["samples"].keys():
+                    uniqSampleName = self.prepend_runid(runId, sample)
+                    if not uniqSampleName in self.bcl2fastq_bysample:
+                        self.bcl2fastq_bysample[uniqSampleName] = {"total": 0, "perfectIndex": 0}
+                    self.bcl2fastq_bysample[uniqSampleName]["total"] += self.bcl2fastq_data[runId][lane]["samples"][sample]["total"]
+                    self.bcl2fastq_bysample[uniqSampleName]["perfectIndex"] += self.bcl2fastq_data[runId][lane]["samples"][sample]["perfectIndex"]
+                    if sample != "undetermined":
+                        if not uniqSampleName in self.source_files:
+                            self.source_files[uniqSampleName] = []
+                        self.source_files[uniqSampleName].append(self.bcl2fastq_data[runId][lane]["samples"][sample]["filename"])
+
+    def add_general_stats(self):
+        data = {key: {"total": self.bcl2fastq_bysample[key]["total"], "perfectPercent": '{0:.1f}'.format(float(100.0*self.bcl2fastq_bysample[key]["perfectIndex"]/self.bcl2fastq_bysample[key]["total"]))} for key in self.bcl2fastq_bysample.keys()}
```

Comment:
So this additional `float` won't actually help, as if they're `int`s then the calculation still comes back as an `int`. Then this call makes that int a float, but still no decimal places.

The `100.0` will force float conversion though, so I think that's fine. And I'm getting decimals now so this is fine 👍